Sources/PT.PM.Matching/Patterns/PatternMethodDeclaration.cs

```diff
@@ -74,7 +74,23 @@ public override MatchingContext Match(Ust ust, MatchingContext context)
                 {
                     if (Body != null)
                     {
-                        newContext = Body.Match(methodDeclaration.Body, newContext);
+                        if (Body is PatternArbitraryDepth || Body is PatternStatements)
+                        {
+                            newContext = Body.Match(methodDeclaration.Body, newContext);
+                        }
+                        else
+                        {
+                            var statements = methodDeclaration.Body.Statements;
```

Comment:
Use more clear name `otherStatements` or something like this for getting rid of confusion between Pattern UST statements and UST statements.